Snowflake.Data.Tests/IntegrationTests/StructuredTypesIT.cs

```diff
@@ -0,0 +1,1471 @@
+                        'FloatValue', 1.23,
+                        'DoubleValue', 1.23,
+                        'DecimalValue', 1.23,
+                        'BooleanValue', true,
+                        'GuidValue', '57af59a1-f010-450a-8c37-8fdc78e6ee93',
+                        'DateTimeValue', '2024-07-11 14:20:05'::TIMESTAMP_NTZ,
+                        'DateTimeOffsetValue', '2024-07-11 14:20:05'::TIMESTAMP_LTZ,
+                        'TimeSpanValue', '14:20:05'::TIME,
+                        'BinaryValue', TO_BINARY('this is binary data', 'UTF-8'),
+                        'SemiStructuredValue', OBJECT_CONSTRUCT('a', 'b')
+                    )::OBJECT(
+                        StringValue VARCHAR,
+                        CharValue CHAR,
+                        ByteValue SMALLINT,
+                        SByteValue SMALLINT,
+                        ShortValue SMALLINT,
+                        UShortValue INTEGER,
+                        IntValue INTEGER,
+                        UIntValue INTEGER,
+                        LongValue BIGINT,
+                        ULongValue BIGINT,
+                        FloatValue FLOAT,
+                        DoubleValue DOUBLE,
+                        DecimalValue REAL,
+                        BooleanValue BOOLEAN,
+                        GuidValue TEXT,
+                        DateTimeValue TIMESTAMP_NTZ,
+                        DateTimeOffsetValue TIMESTAMP_LTZ,
+                        TimeSpanValue TIME,
+                        BinaryValue BINARY,
+                        SemiStructuredValue OBJECT
+                    ), '2024-07-11 14:20:05'::TIMESTAMP_LTZ";
+                    var bytesForBinary = Encoding.UTF8.GetBytes("this is binary data");
+                    command.CommandText = $"SELECT {allTypesObjectAsSFString}";
+                    var reader = (SnowflakeDbDataReader) command.ExecuteReader();
+                    Assert.IsTrue(reader.Read());
+
+                    // act
+                    var allTypesObject = reader.GetObject<AllTypesClass>(0);
+
+                    // assert
+                    Assert.NotNull(allTypesObject);
+                    Assert.AreEqual("abc", allTypesObject.StringValue);
+                    Assert.AreEqual('x', allTypesObject.CharValue);
+                    Assert.AreEqual(15, allTypesObject.ByteValue);
+                    Assert.AreEqual(-14, allTypesObject.SByteValue);
+                    Assert.AreEqual(1200, allTypesObject.ShortValue);
+                    Assert.AreEqual(65000, allTypesObject.UShortValue);
+                    Assert.AreEqual(150150, allTypesObject.IntValue);
+                    Assert.AreEqual(151151, allTypesObject.UIntValue);
+                    Assert.AreEqual(9111222333444555666, allTypesObject.LongValue);
+                    Assert.AreEqual(9111222333444555666, allTypesObject.ULongValue); // there is a problem with 18111222333444555666 value
```

Comment:
What kind of problem?

Reply:
I removed the comment. But json.Value<string>() for large ulong is throwing that "Object must implement IConvertible."